client/src/components/AddProductModal.jsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Button,
+  Grid,
+  makeStyles,
+  Modal,
+  Paper,
+  Slide,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import IconButton from "@material-ui/core/IconButton";
+import React, { useContext } from "react";
+import ArrowBackIcon from "@material-ui/icons/ArrowBack";
+import { userContext as context } from "../providers/UsersProvider";
+
+const useStyles = makeStyles(() => ({
+  paper: {
+    outline: "none",
+    width: "650px",
+    height: "700px",
+    position: "relative",
+    padding: "20px",
+    backgroundColor: "#f2f2f2",
+  },
+  textField: {
+    borderRadius: `4px 4px 4px 4px`,
+    outline: "none",
+    padding: "7px",
+    width: "250px",
+    backgroundColor: "white",
+  },
+  button: {
+    height: 56,
+    borderRadius: 28,
+    width: 142,
+    display: "block",
+    margin: "auto",
+    marginBottom: "20px",
+    marginTop: "20px",
+  },
+  iconButton: {
+    position: "absolute",
+    left: "20px",
+    top: "20px",
+  },
+  newItem: {
+    display: "block",
+    textAlign: "center",
+    margin: "60px 0 15px 0",
+  },
+  linkToItem: {
+    textAlign: "center",
+    marginTop: "50px",
+    fontWeight: "bold",
+  },
+  grid: { width: "fitContent", margin: "20px auto 20px auto" },
+  selectListHeader: {
+    textAlign: "center",
+    marginTop: "50px",
+    fontWeight: "bold",
+  },
+}));
+
+const AddProductModal = () => {
+  const userContext = useContext(context);
+  const classes = useStyles();
+
+  const { isAddingProd, updateIsAddingProd } = userContext;
+
+  const submitHandler = () => {
+    //Use a route for adding a product to list
+  }
+
+  return (
+    <Modal
+      open={isAddingProd}
+      onBackdropClick={() => updateIsAddingProd(false)}
+    >
+      <Slide 
+        direction="left"
+        in={isAddingProd}
+        mountOnEnter
+        style={{ display: "block", margin: "100px auto 0 auto" }} >
```

Comment:
is it possible to make this enter from the side of the dialog itself? Instead of the side of the window?

Reply:
So far I havent found a way but I'm still looking into to it.